autoscale/src/test/java/com/sequenceiq/periscope/endpointtests/DistroXAutoScaleScalingActivityV1EndpointTest.java

```diff
@@ -0,0 +1,485 @@
+                .getFailedScalingActivitiesInGivenDurationByClusterName(TEST_CLUSTER_NAME_1, 60, 0, 3)
+                .stream().map(DistroXAutoscaleScalingActivityResponse::getActivityCrn).collect(Collectors.toList());
+        assertEquals(distroXAutoscaleScalingActivityResponse, List.of());
+    }
+
+    @Test
+    public void testGetFailedScalingActivitiesInGivenDurationByClusterNameWithNullName() {
+        assertThrows(InternalServerErrorException.class, () -> distroXAutoScaleScalingActivityV1Endpoint
+                .getFailedScalingActivitiesInGivenDurationByClusterName(TEST_CLUSTER_NAME_NULL, 60, 0, 10));
+    }
+
+    @Test
+    public void testGetFailedScalingActivitiesInGivenDurationByClusterNameWithNoFailedScalingActivityInGivenRange() {
+        List<String> distroXAutoscaleScalingActivityResponse = distroXAutoScaleScalingActivityV1Endpoint
+                .getFailedScalingActivitiesInGivenDurationByClusterName(TEST_CLUSTER_CRN, 0, 2, 1)
+                .stream().map(DistroXAutoscaleScalingActivityResponse::getActivityCrn).collect(Collectors.toList());
+        assertEquals(distroXAutoscaleScalingActivityResponse, List.of());
+    }
+
+    @Test
+    public void testGetFailedScalingActivitiesInGivenDurationByClusterCrn() {
+        List<String> distroXAutoscaleScalingActivityResponse = distroXAutoScaleScalingActivityV1Endpoint
+                .getFailedScalingActivitiesInGivenDurationByClusterCrn(TEST_CLUSTER_CRN, 600, 2, 1)
+                .stream().map(DistroXAutoscaleScalingActivityResponse::getActivityCrn).collect(Collectors.toList());
+        assertEquals(distroXAutoscaleScalingActivityResponse, List.of(TEST_ACTIVITY_CRN_3));
+    }
+
+    @Test
+    public void testGetFailedScalingActivitiesInGivenDurationByClusterCrnWithNoFailedScalingActivity() {
+        List<String> distroXAutoscaleScalingActivityResponse = distroXAutoScaleScalingActivityV1Endpoint
+                .getFailedScalingActivitiesInGivenDurationByClusterCrn(TEST_CLUSTER_CRN_2, 60, 0, 3)
+                .stream().map(DistroXAutoscaleScalingActivityResponse::getActivityCrn).collect(Collectors.toList());
+        assertEquals(distroXAutoscaleScalingActivityResponse, List.of());
+    }
+
+    @Test
+    public void testGetFailedScalingActivitiesInGivenDurationByClusterCrnWithNoScalingActivityStored() {
+        List<String> distroXAutoscaleScalingActivityResponse = distroXAutoScaleScalingActivityV1Endpoint
+                .getFailedScalingActivitiesInGivenDurationByClusterCrn(TEST_CLUSTER_CRN_1, 60, 0, 3)
+                .stream().map(DistroXAutoscaleScalingActivityResponse::getActivityCrn).collect(Collectors.toList());
+        assertEquals(distroXAutoscaleScalingActivityResponse, List.of());
+    }
+
+    @Test
+    public void testGetFailedScalingActivitiesInGivenDurationByClusterCrnWithNullCrn() {
+        assertThrows(InternalServerErrorException.class, () -> distroXAutoScaleScalingActivityV1Endpoint
```

Comment:
Same here. Ensure that `BadRequestException` is the one that's thrown and replace this accordingly